GoMeet/src/test/java/com/google/sps/LocationServletTest.java

```diff
@@ -75,51 +78,39 @@ public void tearDown() {
   }
 
   /** 
-   * Tests if new location entity is accurately added to the Database.
-   * Run this test twice to prove we're not leaking any state across tests.
+   * Tests if the recieved requests is correctly sent to the LocationDAO
+   * and the key is sent in the response.
    */
-  private void doPostTest() throws IOException {
-    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
-   
-    when(request.getParameter("title")).thenReturn(TITLE_A);
-    when(request.getParameter("lat")).thenReturn(LAT_A);
-    when(request.getParameter("lng")).thenReturn(LNG_A);
-    when(request.getParameter("note")).thenReturn(NOTE_A);
-
+  @Test
+  public void doPostTest() throws IOException {
+    // Set up request mock
+    when(request.getParameter("title")).thenReturn(LOCATION_A.getTitle());
+    when(request.getParameter("lat")).thenReturn(Double.toString(LOCATION_A.getLat()));
+    when(request.getParameter("lng")).thenReturn(Double.toString(LOCATION_A.getLng()));
+    when(request.getParameter("note")).thenReturn(LOCATION_A.getNote());
+
+    // Set up response mock writer
     StringWriter stringWriter = new StringWriter();
```

Comment:
Could potentially define `stringWriter` and `writer` as private instance variables, so you don't have to set up the response mock writer in both test cases separately.

Reply:
Yup, sounds good. Will do